modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalTestHelper.java

```diff
@@ -42,4 +49,20 @@ public static XContentParser roundtrip(ToXContentToBytes testItem) throws IOExce
         XContentParser itemParser = XContentHelper.createParser(shuffled.bytes());
         return itemParser;
     }
+
+    public static void testHashCodeAndEquals(Object testItem, Object mutation, Object secondCopy) {
```

Comment:
I thought about this a bit more and opened https://github.com/elastic/elasticsearch/issues/20629 for discussion.